### ISSUE TITLE
Check PHP version before trying to create PHP server

### ISSUE THREAD

**Issue by JoshHarmon** [pkill37/mybb-shetup#1]
We should check the user's PHP version before running the PHP server.

If it's less than 5.4, we could just echo a message. Otherwise, open the server.

We'll need to do something like PHP_VERSION=`php -v` and find the version number within.

**Comment by pkill37** [comment]
We can use `php -v` or run either `phpinfo()` or the `PHP_VERSION` constant through `php -r` to get the PHP version. It's not too difficult to do so, but the version number output is different on many machines. It could be simply 5.4.6 or, in my case, 5.4.6-1ubuntu1.2.

This doesn't seem to make a difference considering the above two cases. But it's possible other machines output the PHP version a bit differently. Anyway, I have the following three solutions:

``` bash
$ php -v | head -n 1 | cut -d ' ' -f2 | cut -d '-' -f1

$ php -r \@phpinfo\(\)\; | grep 'PHP Version' -m 1 | cut -d ' ' -f4 | cut -d '-' -f1

$ php -r echo\\PHP_VERSION\; | cut -d ' ' -f4 | cut -d '-' -f1
```

The `cut -d '-' -f1` thing splits up 5.4.6-1ubuntu1.2 for example and takes the stuff on the left side of the hyphen. Luckily this is ignored if you already got 5.4.6 or whatever. I believe you're running OSX. Would you mind testing this on your machine? Any cases in which this might not work?

If this works, then we just have to check if the number after the first dot is a 4 to ensure the user is running PHP 5.4.

**Comment by JoshHarmon** [comment]
First and second work great on my Mac (echoes 5.3.15). I think the third works too, seeing as it did echo the PHP version. However, it didn't output a newline at the end, FWIW.

**Comment by pkill37** [comment]
Thanks for the input.

I think `php -v` may be the best solution. We'll come back to this later if it causes any issues.

**Comment by pkill37** [comment]
I've been wondering how we should handle requirements exactly. Should we check the requirements as we go (in this case before starting the PHP server) or introduce some sort of a requirements check in the beginning that would fail unless all requirements are met?

I think the former makes more sense because the script only really becomes useful if you take full advantage of it. Any input on this?

**Comment by JoshHarmon** [comment]
I don't think we should _require_ PHP 5.4. Sure it's stable, but a lot of people haven't moved to it yet.  

But I think that a check at the start would probably be the best for the script.  

## 

Josh Harmon
Sent with Sparrow (http://www.sparrowmailapp.com/?sig)

**Comment by pkill37** [comment]
PHP 5.4 is what makes the open browser feature possible though. Anything below that won't work.

What you said is true for production, but I like to think developers have moved to 5.4 already. And if they haven't they shouldn't have a problem upgrading.

So you're okay doing a full check on requirements and aborting unless all requirements are met?

**Comment by JoshHarmon** [comment]
For the development version, that's fine. But for production, I don't think we should.

Now that we have base code, I think we're almost ready to push the development version to master and production to a new branch. We should fix the POSIX issues first though.

**Comment by pkill37** [comment]
Closing this. Will open a separate issue for checking requirements.
